sdk/src/main/java/com/horizen/cryptolibprovider/utils/FieldElementUtils.java

```diff
@@ -11,6 +11,14 @@ public static int fieldElementLength() {
         return Constants.FIELD_ELEMENT_LENGTH();
     }
 
+    public static FieldElement messageToFieldElementStrict(byte[] message) {
+        if (message.length != fieldElementLength()) {
+            throw new IllegalArgumentException("Message length is exceed allowed message len. Message len " +
+                    message.length + " but it shall be equal to " + fieldElementLength());
+        }
+        return FieldElement.deserialize(message);
+    }
+
     public static FieldElement messageToFieldElement(byte[] message) {
```

Comment:
If I am not wrong we agreed to change this method name because it is not meant anymore to deal with messages to sign. And also to change the variable names in the code of the method `generateHashAndCleanUp()` in `sdk/src/main/scala/com/horizen/consensus/package.scala`

Reply:
Renamed message to element